src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsallocations/client/AssessRisksNeedsApiClient.kt

```diff
@@ -1,49 +1,35 @@
 package uk.gov.justice.digital.hmpps.hmppsallocations.client
 
+import kotlinx.coroutines.flow.Flow
 import org.springframework.http.HttpStatus
 import org.springframework.web.reactive.function.client.WebClient
-import reactor.core.publisher.Flux
+import org.springframework.web.reactive.function.client.awaitBodyOrNull
+import org.springframework.web.reactive.function.client.bodyToFlow
 import reactor.core.publisher.Mono
 import uk.gov.justice.digital.hmpps.hmppsallocations.domain.RiskPredictor
 import uk.gov.justice.digital.hmpps.hmppsallocations.domain.RoshSummary
 
 class AssessRisksNeedsApiClient(private val webClient: WebClient) {
 
-  fun getRosh(crn: String): Mono<RoshSummary> {
+  suspend fun getRosh(crn: String): RoshSummary? {
     return webClient
       .get()
       .uri("/risks/crn/$crn/widget")
       .retrieve()
       .onStatus(
         { httpStatus -> HttpStatus.NOT_FOUND == httpStatus },
-        { Mono.error(MissingRiskError("No risk summary found for $crn")) }
-      )
-      .bodyToMono(RoshSummary::class.java)
-      .onErrorResume { ex ->
-        when (ex) {
-          is MissingRiskError -> Mono.empty()
-          else -> Mono.error(ex)
-        }
-      }
+        { Mono.empty() }
+      ).awaitBodyOrNull()
   }
 
-  fun getRiskPredictors(crn: String): Flux<RiskPredictor> {
+  fun getRiskPredictors(crn: String): Flow<RiskPredictor> {
```

Comment:
Doesn't this need a `suspend`?

Reply:
`bodyToFlow()` isn't a suspend function so the compiler said suspend was redundant